test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -2611,8 +2611,6 @@ def forward(self, x):
         x = torch.empty(2, 3, 3, dtype=torch.double).uniform_(0, 1)
         self.run_test(Bernoulli(), x)
 
-    # Enable test when fix for allowzero is in ORT
-    @skipForAllOpsetVersions()
     @skipIfUnsupportedMinOpsetVersion(14)
     def test_reshape_allowzero(self):
         class ReshapeModel(torch.nn.Module):
```

Comment:
I think safeguard was added to protect against ``` test_reshape_different_rank```
This is a scenario where:
``` 
allowzero = 1
shape has -1
shape has no 0
```
Ideally torch should only fail for cases where:
``` 
allowzero = 1
shape has -1
shape has 0
0 and -1 cannot both be present in `Shape` input of `Reshape` node, when `allowzero=1`
```
and not for the case above.

However as seen in ORT https://github.com/microsoft/onnxruntime/blob/7660eeef3eff9a58b4d807dd951a34bafcb03ac0/onnxruntime/core/providers/cpu/tensor/reshape_helper.h#L20
ORT is enforcing that if ```allowzero=1```, then ```shape cannot have a -1```
This check need to be extended such that  if ```allowzero=1```, then ```shape cannot have a -1 and 0```